instructor/compiler.py

```diff
@@ -406,13 +401,13 @@ def gel_types(a):
 
 
 def micro_kinds(a):
-    ret = {}
-    for sp in a.microscopy_sample_prep.all():
-        analysis = sp.analysis
-        condition = sp.condition
+    ret = {'na': {'name': 'None'}}
+    for sample_prep in a.microscopy_sample_prep.all():
+        analysis = sample_prep.micro_analysis
+        condition = sample_prep.condition
         if analysis not in ret:
             ret[analysis] = {
-                'name': analysis,
+                'name': sample_prep.get_micro_analysis_display(),
```

Comment:
it is built in https://docs.djangoproject.com/en/1.10/ref/models/instances/#extra-instance-methods